dor/cli/upload.py

```diff
@@ -39,38 +44,55 @@ async def _run_upload(
     if not type_profiles:
         typer.echo("No profiles provided. Use --image or --text to specify profiles.")
         raise typer.Exit(1)
+
     profiles = generate_profiles(folder_path=Path(folder), type_profiles=type_profiles)
-    fileset_profiles: dict[str,dict] = {}
+    fileset_profiles: dict[str, dict[str, list[str]]] = {}
     for file_name in profiles:
         name = Path(file_name).stem
         if name not in fileset_profiles:
-            fileset_profiles[name] = {}    
-        fileset_profiles[name][file_name] = profiles[file_name]    
-    for name in fileset_profiles:             
-        try:
-            async with httpx.AsyncClient(follow_redirects=True) as client:
-                result = await run_upload_fileset(
-                    client,
-                    base_url,
-                    profiles=fileset_profiles[name],
-                    folder=folder,
-                    name=name,
-                    project_id=project_id,                   
-                )
-            typer.echo(f"Fileset created successfully: {result}")
+            fileset_profiles[name] = {}
+        fileset_profiles[name][file_name] = profiles[file_name]
 
-        except UploadError as exc:
-            typer.secho(f"Upload error: {exc}", fg="white", bg="red", bold=True)
-            typer.echo(f"Details: {exc.message}, Code: {exc.code}", err=True)
-            raise typer.Exit(1)
+    semaphore = asyncio.Semaphore(10)
 
-        except httpx.RequestError as exc:
-            typer.echo(f"An error occurred while making the request: {exc}", err=True)
-            raise typer.Exit(1)
+    async with httpx.AsyncClient(follow_redirects=True) as client:
+        tasks = []
+        for name in fileset_profiles:
+            tasks.append(run_upload_fileset_with_limit(
+                sempahore=semaphore,
+                client=client,
+                base_url=base_url,
+                project_id=project_id,
+                name=name,
+                profiles=fileset_profiles[name],
+                folder=folder,
+            ))
+        results = await asyncio.gather(*tasks, return_exceptions=True)
 
-        except httpx.HTTPStatusError as exc:
+    data_results = []
+    exceptions = []
+    for result in results:
+        if isinstance(result, Exception):
+            exceptions.append(exceptions)
+        else:
+            data_results.append(result)
+
+    typer.echo(f"Successfully submitted {len(data_results)} fileset(s) for creation.")
+    rich.print(data_results)
```

Comment:
It seems useful for now to print the response data for each request here, but at scale, this isn't practical. How should we handle this? We could not print anything, or only print if the length of results is at or below a maximum size?

Reply:
https://github.com/mlibrary/project-dor-docs/discussions/89#discussioncomment-13286953